### DISCUSSION TITLE
Load script on demand

### DISCUSSION THREAD

**Discussion by urbanambroz** [gtm-support/vue-gtm#330] [Q&A] [answered]
Since the loadScript boolean is available to be set to false I think its a good idea to expose the loadScript() function call as well or rather have it invokable on demand.

A use case for this would be the case that we do not want to load the script unless the correct cookie consent was given.

Currently the only alternative I see is setting loadScript to false and then manually adding the GTM script when we decide to but this step is redundant when using You library and I think adding support for it would be a nice thing.

**Comment by Shinigami92** [comment, marked as answer]
From the README:

> `enabled: true, // defaults to true. Plugin can be disabled by setting this to false for Ex: enabled: !!GDPR_Cookie (optional)`

This is exactly your use-case

So later when reading the cookie and it is accepted, then you can call `gtm.enable()` and you have exactly what you request

**Reply by urbanambroz** [reply]
Whoops You're right, and I saw that but thought that loading is something else than just disabling it. I apologize for the uselessness of the question :)
